syntax = "proto3";

package NetProto;

enum Type {
  HEADER_TYPE_UNKNOWN = 0;
  HEADER_TYPE_FRAME = 1;
  HEADER_TYPE_UPDATE = 2;

  // Reliable messages
  HEADER_TYPE_START_SENDING_FRAMES = 1001;

  HEADER_TYPE_SET_AS = 1010;
  HEADER_TYPE_LOAD_SCENE = 1011;

  HEADER_TYPE_REQUEST_FILE_ENTRIES = 1020;
  HEADER_TYPE_REQUEST_DIR_ENTRIES = 1021;

  HEADER_TYPE_RETURN_FILE_ENTRIES = 1030;
  HEADER_TYPE_RETURN_DIR_ENTRIES = 1031;
}

message Header {
  Type type = 1;
  uint32 message_size = 2;
}
